docs/operations/gotchas.md

```diff
@@ -0,0 +1,306 @@
+## Gotchas and Rough Spots
+
+Known issues, limitations, and things that might trip you up. This is an early release - some rough edges are still being smoothed out.
+
+---
+
+## Architecture and Design Limitations
+
+### Tool provider configuration per instance is clunky
+Each Redis instance may need specific configuration for tool providers (e.g., different Prometheus label selectors, different Loki tenant IDs). Currently:
+- Store these in the instance's `extension_data` (tool provider config)
+- Store secrets in `extension_secrets` (tool provider secrets)
+- Access via instance API or CLI
+
+**The problem**: This works but isn't elegant. You're managing per-instance config in JSON blobs rather than a structured interface.
+
+**Future direction**: We're exploring better patterns for per-instance tool configuration, possibly with provider-specific schemas and validation.
+
+### Envelope encryption uses environment variable for master key
+The agent uses envelope encryption to protect secrets (connection URLs, passwords, API keys) stored in Redis. The encryption works like this:
+1. Master key (32-byte secret) is stored in `REDIS_SRE_MASTER_KEY` environment variable
+2. Data encryption keys (DEKs) are generated per secret
+3. DEKs are encrypted with the master key and stored alongside encrypted data
+
+**The problem**: The master key is an environment variable, not integrated with a secrets store (Vault, AWS Secrets Manager, etc.). For some users, this is a non-starter.
+
+**Workarounds**:
+- Use your orchestration system's secrets management (Kubernetes Secrets, systemd credentials, etc.) to inject the env var
+- Rotate the master key periodically (requires re-encrypting all secrets)
+
+**Future direction**: Direct integration with secrets stores is on the roadmap.
+
+### AI agent connecting to production databases is risky
+The Redis CLI diagnostics provider is useful (especially if you're not already scraping `INFO` metrics), but it requires storing the target database's connection URL. This means:
+- The agent can connect directly to production Redis instances
```

Comment:
The agent can never run commands directly on the target instance. It can only use the tool provider interface, which exposes read-only commands indirectly.